xla/stream_executor/cuda/assemble_compilation_provider_test.cc

```diff
@@ -37,6 +39,18 @@ using ::testing::AllOf;
 using ::testing::HasSubstr;
 using ::tsl::testing::StatusIs;
 
+TEST(AssembleCompilationProviderTest,
+     CandidateCudaRootsConsidersCUDA_HOME) {
+  const std::string cuda_home = "/my/cuda/home";
+  tsl::setenv("CUDA_HOME", cuda_home.c_str(), 1);
+  auto roots = tsl::CandidateCudaRoots();
+  if (roots.empty()) {
+    GTEST_SKIP() << "Test only when any root is set";
+  }
+  EXPECT_THAT(std::find(roots.begin(), roots.end(), cuda_home),
+              Ne(roots.end()));
```

Comment:
Can you use `EXPECT_THAT(roots, ::testing::Contains(cud_home));` here?

This will produce a better error message when it fails.